R/tree.R

```diff
@@ -31,200 +31,284 @@
-    # read in sequence data, convert to phyDat
-    prot_fa <- read.phyDat(fa_file, format = "fasta", type = "AA")
-    prot_phyDat <- phyDat(prot_fa, type = "AA", levels = NULL)
-    prot10 <- subset(prot_phyDat, 1:10)
-    prot10_phyDat <- phyDat(prot10, type = "AA", levels = NULL)
+        ###########################
+        ## Approach 2
+        ###########################
+        ## Alignment file formats and conversion
+        # read in sequence data, convert to phyDat
+        prot_fa <- read.phyDat(fa_file, format = format, type = seq_type)
+        prot_phyDat <- phyDat(prot_fa, type = seq_type, levels = NULL)
+        prot_subset <- subset(prot_phyDat, 1:subset_size)
+        prot_subset_phyDat <- phyDat(prot_subset, type = seq_type, levels = NULL)
 
-    ## Model Testing & Distance Matrices
-    ## Comparison of different nucleotide or amino acid substitution models
-    mt <- modelTest(prot10, model = "all")
-    print(mt)
+        ## Model Testing & Distance Matrices
+        ## Comparison of different nucleotide or amino acid substitution models
+        mt <- modelTest(prot_subset, model = model_test)
+        print(mt)
 
-    # estimate a distance matrix using a Jules-Cantor Model
-    dna_dist <- dist.ml(prot10, model = "JC69")
+        # estimate a distance matrix using a Jules-Cantor Model
+        dna_dist <- dist.ml(prot_subset, model = dna_model)
 
-    ## Neighbor Joining, UPGMA, and Maximum Parsimony
-    # quick and dirty UPGMA and NJ trees
-    prot_UPGMA <- upgma(dna_dist)
-    prot_NJ <- NJ(dna_dist)
-    plot(prot_UPGMA, main = "UPGMA")
-    plot(prot_NJ, main = "Neighbor Joining")
+        ## Neighbor Joining, UPGMA, and Maximum Parsimony
+        # quick and dirty UPGMA and NJ trees
+        prot_UPGMA <- upgma(dna_dist)
+        prot_NJ <- NJ(dna_dist)
+        plot(prot_UPGMA, main = "UPGMA")
+        plot(prot_NJ, main = "Neighbor Joining")
 
-    # parsimony searches
-    prot_optim <- optim.parsimony(prot_NJ, prot)
-    prot_pratchet <- pratchet(prot10) # returning error
-    plot(prot_optim)
-    plot(prot_pratchet)
+        # parsimony searches
+        prot_optim <- optim.parsimony(prot_NJ, prot_phyDat)
+        prot_pratchet <- pratchet(prot_subset) # returning error
+        plot(prot_optim)
+        plot(prot_pratchet)
 
-    ## Maximum likelihood and Bootstrapping
-    # ml estimation w/ distance matrix
-    fit <- pml(prot_NJ, prot10)
-    print(fit)
-    fitJC <- optim.pml(fit, model = "JC", rearrangement = "stochastic")
-    logLik(fitJC)
-    bs <- bootstrap.pml(fitJC,
-        bs = 100, optNni = TRUE, multicore = TRUE,
-        control = pml.control(trace = 0)
-    )
-    plotBS(midpoint(fitJC$tree), bs, p = 50, type = "p")
+        ## Maximum likelihood and Bootstrapping
+        # ml estimation w/ distance matrix
+        fit <- pml(prot_NJ, prot_subset)
+        print(fit)
+        fitJC <- optim.pml(fit, model = ml_model, rearrangement = rearrangement)
```

Comment:
```suggestion
        fit_optimized <- optim.pml(fit, model = fit, rearrangement = rearrangement)
```

Or something similar. JC here stands for Jukes-Cantor, which is a DNA substitution model, so removing that from the variable. I believe the model in `model = ml_model` will actually be the prior tree generated by `pml()`, which would be `fit` here, and then `optim.pml()` optimizes the previous model. Do double-check me on this, though!

Reply:
Yes, switching model based on DNA/protein input could be part of the function, too.